src/Model/Behavior/SlugBehavior.php

```diff
@@ -194,11 +197,12 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
         $fields = (array)$config['displayField'];
         $parts = [];
         foreach ($fields as $field) {
-            if (!isset($entity->{$field}) && !$entity->isNew()) {
+            $value = Hash::get($entity, $field);
+
+            if ($value === null && !$entity->isNew()) {
```

Comment:
Can't we support both? I mean checking Cake version and enabling dot-notation support for slug or not otherwise. Not sure we should drop support for 3.0 if we can avoid it. What do you think?